compiler/scopelang/dependency.ml

```diff
@@ -19,7 +19,46 @@
 
 open Catala_utils
 open Shared_ast
-module SVertex = ScopeName
+
+type vertex = Scope of ScopeName.t | Topdef of TopdefName.t
+
+module SVertex = struct
+  type t = vertex
+  (* While we enforce that globals don't depend on scopes, and could therefore
```

Comment:
Well do we really enforce this invariant right now? Direct scope calling is an expression so syntactically it could fit in a toplevel def right?

Reply:
Syntactically, it could, but wereject it explicitely [during desugaring](https://github.com/CatalaLang/catala/pull/391/files/c3af0b409735d4fff803b40fc9a19f37809641f1#diff-185db1a11ce406fa3ae293f0de69e5252be36e4f312c5195ad6216c93c73d43bR403-R406)